SingleLinkedList.java

```diff
@@ -0,0 +1,157 @@
+////////////////////////////////////////////////////////////////////////////////
+// Author: Ben Kuschner
+////////////////////////////////////////////////////////////////////////////////
+
+public class SingleLinkedList<T> {
+   
+   /////////////////////////////////////////////////////////
+   // Private link class
+   /////////////////////////////////////////////////////////
+   private class Link<T> {
+      // Member Variables
+      private T data;
+      private Link<T> next;
+
+      // Constructor(s)
+      public Link (T data) {
+         this.data = data;
+         next = null;
+      }
+   
+      // Member methods
+      public T getData() {
+         return this.data;
+      }
+      
+      //deleted hasNext() method because I did not use it
+   
+      public Link<T> getNext() {
+         return next;
+      }
+
+      public void setNext(Link<T> next) {
+         this.next = next;
+      }
+
+   } // end of class link
+
+   /////////////////////////////////////////////////////////
+   // Single Linked List
+   /////////////////////////////////////////////////////////
+
+   // Member variables
+   private Link<T> first;
```

Comment:
Small note, convention would refer to the first node in the linked list as the head, and the last as the tail.